widget/controllers/widget.item.controller.js

```diff
@@ -248,6 +248,14 @@
               WidgetItem.item.isRedeemed = true;
               WidgetItem.item.redeemedOn = result.data.redeemedOn;
               $scope.enableRedeemButton();
+              var currentView = ViewStack.getCurrentView();
+
+              if (currentView.params && currentView.params.itemId && !currentView.params.stopSwitch) {
+                //Track Coupon Plugin Event
+                buildfire.analytics.trackAction("coupon_item_view_" + currentView.params.itemId+"_redeemed");
```

Comment:
I don't think you have the right key here. When you did event registration, you said it will be 

ContentItem.item.id+"_redeemed"  

but here, it's 

"coupon_item_view_" + currentView.params.itemId+"_redeemed"

Am I missing something?

Reply:
in the track action function, the currentView.params.itemId is actually just the id for example "1551" because the currentview returns the page that the user is on where the coupon is an element with a specific id but when registering an item its registered it as coupon_item_view_{the id of the item}_redeemed.
I did notice that i changed a key that was already registered, so I am going to revert it back and push the updates